imagenet/main.py

```diff
@@ -71,6 +71,8 @@
                     help='seed for initializing training. ')
 parser.add_argument('--gpu', default=None, type=int,
                     help='GPU id to use.')
+parser.add_argument("--xpu", default=False, type=bool,
```

Comment:
It seems architecture of this example was to automatically detect the device to use. I would not step out from that. If you want a way to disable XPU (might make sense to still be able to run on CPU), add `--no-xpu` option.